src/transaction_sender/service.zig

```diff
@@ -0,0 +1,302 @@
+const std = @import("std");
+const network = @import("zig-network");
+const sig = @import("../sig.zig");
+
+const socket_utils = sig.net.socket_utils;
+
+const AtomicBool = std.atomic.Value(bool);
+const AtomicSlot = std.atomic.Value(Slot);
+const UdpSocket = network.Socket;
+
+const SocketAddr = sig.net.SocketAddr;
+const Packet = sig.net.Packet;
+const Slot = sig.core.Slot;
+const RwMux = sig.sync.RwMux;
+const Signature = sig.core.Signature;
+const Channel = sig.sync.Channel;
+const Instant = sig.time.Instant;
+const GossipTable = sig.gossip.GossipTable;
+const RpcClient = sig.rpc.Client;
+const Timer = sig.time.Timer;
+const Logger = sig.trace.log.Logger;
+const Config = sig.transaction_sender.Config;
+const Stats = sig.transaction_sender.Stats;
+const LeaderInfo = sig.transaction_sender.LeaderInfo;
+const TransactionInfo = sig.transaction_sender.TransactionInfo;
+const TransactionPool = sig.transaction_sender.TransactionPool;
+
+/// Basic send transaction service that listens for transactions on a channel.
+/// Transactions are added to the pool and retried until they are confirmed, failed,
+/// timed out, or reach the max retries.
+/// The leader schedule and current slot are loaded via RPC calls to the cluster.
+/// The leader TPU addresses are loaded from the gossip table.
+/// TODO:
+/// - Add logging
+/// - Add stats tracking
+/// - Add nonce handling
+/// - Remove RPC calls
+pub const Service = struct {
+    allocator: std.mem.Allocator,
+    config: Config,
+    stats: Stats,
```

Comment:
do you think stats and config need their own files? can just have them defined after service in this file like we do in gossip